### PR TITLE
Fixing kubemark logs

### PR DESCRIPTION
**Pull request by krzysied**
- splitting `kubemarkTest` into `kubemarkUp` and `kubemarkGinkgoTest`
- moving log dumping to `kubemarkDown`

ref #11518

### CODE REVIEW

**Comment by krzysied** [comment]
/assign @wojtek-t 
/cc @BenTheElder @krzyzacy 

I've done a little refactor here:
- `kubemarkUp` - sets kubemark and required env varibiables.
- `kubemarkGinkgoTest` - runs ginkgo tests.
- `kubemarkDown` - dumps logs and tears kubemark down.

**Comment by krzyzacy** [comment]
looks like all kubemark logic can just break into it's own package? :smirk: 

I'll leave to @wojtek-t

File: kubetest/e2e.go [hunk 1]
```diff
@@ -220,11 +226,8 @@ func run(deploy deployer, o options) error {
 
 	// TODO(bentheelder): consider remapping charts, etc to testCmd
 
-	var kubemarkWg sync.WaitGroup
-	var kubemarkDownErr error
 	if o.kubemark {
-		kubemarkWg.Add(1)
-		go kubemarkDown(&kubemarkDownErr, &kubemarkWg)
+		errs = util.AppendError(errs, kubemarkDown(dump))
```
Comment (wojtek-t) [inline kubetest/e2e.go]: This broke the optimization that we tear down kubemark in parallel with tearing down the main cluster. - please revert the WaitGroup.

File: kubetest/e2e.go [hunk 2]
```diff
@@ -565,7 +564,7 @@ func nodeTest(nodeArgs []string, testArgs, nodeTestArgs, project, zone string) e
 	return control.FinishRunning(exec.Command("go", runner...))
 }
 
-func kubemarkTest(testArgs []string, dump string, o options, deploy deployer) error {
+func kubemarkUp(dump string, o options, deploy deployer) error {
```
Comment (wojtek-t) [inline kubetest/e2e.go]: +1 to @krzyzacy suggestion to move all kubemark-related function to a separate file.

Comment (wojtek-t) [reply kubetest/e2e.go]: Though maybe a separate PR, to avoid mixing logic changes with mechanical ones.

File: kubetest/e2e.go (hunk 1 above)
Comment (krzysied) [reply kubetest/e2e.go]: Done.

**Comment by wojtek-t** [comment]
/lgtm
/approve
/hold

@krzyzacy or @BenTheElder for final approval

**Comment by krzyzacy** [comment]
/approve

yeah once this is moved to its own package it'll be easier migrate to kubetest2 :-)

**Pull request merged**